containers/src/Data/IntMap/Internal.hs

```diff
@@ -518,6 +518,12 @@ instance NFData a => NFData (IntMap a) where
     rnf (Tip _ v) = rnf v
     rnf (Bin _ l r) = rnf l `seq` rnf r
 
+-- | @since 0.7.1
+instance NFData1 IntMap where
+    liftRnf _ Nil = ()
+    liftRnf rnfx (Tip _ v) = rnfx v
+    liftRnf rnfx (Bin _ l r) = liftRnf rnfx l `seq` liftRnf rnfx r
+
```

Comment:
Should we use a helper function to allow this to inline?

Reply:
I've added one here - but possibly there are other instances that could benefit?

Also not sure if there's any need for any `INLINE` or `INLINABLE` pragmas.

Reply:
I probably wouldn't add pragmas for this. The function is small enough that I'd expect GHC to do a decent job of deciding when to inline it. Letting others inline would probably be good too.

Reply:
`Seq` will be tricky to allow inlining for (it requires a GADT, or better a fake GADT, and I don't *think* we have one here), so don't worry too much about that.

Reply:
I think that's now done for `Map`, `Set`, `IntMap` and `Tree`.

Reply:
Seems like it would be a good to have a `()`-like type with a strict `<>` instance, then we can write all `NFData` stuff simply using `foldMap` and `foldMapWithKey` instead of rewriting tree traversals every time.

Reply:
@meooow25 , that sounds clever.

Reply:
I wanted to see if this exists in/could be added to `deepseq`, and it seems like you made an attempt a while ago: https://github.com/haskell/deepseq/pull/18 😄
It would be handy to have the types `UnitLTR`/`UnitRTL` available from `deepseq`, but for now we'll have to define our own if we do this.

Reply:
Opened https://github.com/haskell/deepseq/issues/105